### PR TITLE
Project banking support

### PR DESCRIPTION
**Pull request by AmandaBirmingham**
Supports fixing microsetta-admin repo's issues 16 and 18.

This PR adds functionality to the private api's admin endpoints to (a) pass into project creation info on whether or not a project should be banked and if so until what (optional) plating start date and (b) adds this banked-or-not and plating start date info to the project info returned when searching for info on a barcode.  Tests are updated/added to exercise this functionality.

It also changes the /api/admin/scan/{barcode} endpoint that updates sample info on scanning from a post to a put (and changes the http return code from 201 to 200) because this endpoint isn't creating a new record but rather updating an existing one.

### CODE REVIEW

File: microsetta_private_api/api/microsetta_private_api.yaml [hunk 1]
```diff
@@ -990,8 +992,9 @@ paths:
           $ref: '#/components/responses/401Unauthorized'
         '404':
           $ref: '#/components/responses/404NotFound'
+
   '/admin/scan/{sample_barcode}':
-    post:
+    put:
```
Comment (wasade) [inline microsetta_private_api/api/microsetta_private_api.yaml:997]: We'll need to make sure a deploy with this commit is timed with an updated deploy of microsetta-admin

File: microsetta_private_api/api/tests/test_admin_api.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+        # execute project post (create)
+        response = self.client.post(
+            "/api/admin/create/project",
+            content_type='application/json',
+            data=input_json,
+            headers=MOCK_HEADERS
+        )
+
+        # NB: the date format SHOULD be checked by connexion
```
Comment (wasade) [inline microsetta_private_api/api/tests/test_admin_api.py:177]: ugh!! should we `try/except` around `datetime.strptime` prior to the sql query?

File: microsetta_private_api/repo/admin_repo.py [hunk 3]
```diff
@@ -148,10 +157,26 @@ def create_project(self, project_name, is_microsetta):
...
+    def delete_project_by_name(self, project_name):
+        """Delete a project entry in the database
+
+        Parameters
+        ----------
+        project_name : str
+            The name of the project to delete
+        """
+        with self._transaction.cursor() as cur:
+            cur.execute("DELETE FROM barcodes.project WHERE project = %s",
```
Comment (wasade) [inline microsetta_private_api/repo/admin_repo.py:176]: Doesn't this trigger a FK issue with the many-many join table `barcodes.project_barcode`?

Comment (AmandaBirmingham) [reply microsetta_private_api/repo/admin_repo.py:184]: Good catch--I didn't see that bc I only wrote this for use by tests, and the tests I was using it for didn't include any barcode associations!  Added code to delete the barcode associations first ...

File: microsetta_private_api/api/tests/test_admin_api.py (hunk 2 above)
Comment (AmandaBirmingham) [reply microsetta_private_api/api/tests/test_admin_api.py]: Added try/except

File: microsetta_private_api/api/microsetta_private_api.yaml (hunk 1 above)
Comment (AmandaBirmingham) [reply microsetta_private_api/api/microsetta_private_api.yaml:997]: Correct

**Pull request merged**